src/wampy.js

```diff
@@ -897,24 +896,22 @@ class Wampy {
         const needNoSession = [WAMP_MSG_SPEC.WELCOME, WAMP_MSG_SPEC.CHALLENGE].includes(messageType);
         const needValidSession = !needNoSession && messageType !== WAMP_MSG_SPEC.ABORT;
 
-        if (needNoSession && this._cache.sessionId) {
-            return this._hardClose(errorURI, `Received message "${messageType}" after session was established`);
-        }
-
-        if (needValidSession && !this._cache.sessionId) {
-            return this._hardClose(errorURI, `Received message "${messageType}" before session was established`);
+        if (needNoSession && this._cache.sessionId || needValidSession && !this._cache.sessionId) {
+            const relativeTime = needNoSession ? 'after' : 'before';
+            return this._hardClose(errorURI, `Received message "${messageType}" ${relativeTime} session was established`);
         }
 
         await handler();
     }
 
     /**
      * Handles websocket welcome message event
-     * WAMP SPEC: [WELCOME, Session|id, Details|dict]
-     * @param {Array} [, sessionId, details] - decoded event data
+     *
+     * WAMP SPEC: [Session|id, Details|dict]
```

Comment:
I don't like that you removed an item. And now this is actually not a WAMP SPEC. Rather, it is only misleading. 
I think we should preserve what exactly the WAMP message looks like. And just maybe add a small note in addition if you'd like (or even omit, as for me it looks ok)

Reply:
Now that the function parameters are more explicit [sessionId, details] instead of data[1] and data[2] what do you think of removing the WAMP SPEC commentary from the function headers ?

Reply:
Well... I like them there. I work much with the WAMP specification itself and other implementations and it is really convenient for me to see the original WAMP message structure regardless of having most (but not all) of the parameters as function arguments. Also, this helps to correctly name and cross-reference function parameters with the WAMP message options in place. So I would like to keep them.

Reply:
Hmmm okay. I guess I'll close this then to avoid confusion with the actual WAMP specification since it kinda deviates too much from it.